src/components/OneToDo/OneToDo.jsx

```diff
@@ -1,72 +1,98 @@
-import {
-  Checkbox,
-  IconButton,
-  TextField,
-  Typography,
-} from "@mui/material";
+import { Checkbox, IconButton, TextField, Typography } from "@mui/material";
 import DeleteIcon from "@mui/icons-material/Delete";
 import EditIcon from "@mui/icons-material/Edit";
 import SaveIcon from "@mui/icons-material/Save";
-import React, { useEffect } from "react";
+import React, { useRef } from "react";
 import { useState } from "react";
 import "./OneToDo.css";
+import { useDispatch, useSelector } from "react-redux";
+import {
+  reduxEditToDo,
+  reduxDeleteToDo,
+  toggleCheck,
+} from "../../redux/toDoSlice/toDoSlice";
 
-function OneToDo({ toDo, editToDo, deleteToDo, index }) {
+function OneToDo({ index }) {
   const [editingToDo, setEditingToDo] = useState(false);
-  const [editField, setEditField] = useState("");
   const [toDoStatus, setToDoStatus] = useState(false);
 
-  useEffect(() => {
-    setEditField(toDo);
-  }, [toDo]);
+  const dispatch = useDispatch();
+
+  const reduxToDoList = useSelector((state) => {
+    return state.toDo.toDoList;
+  });
+
+  const editField = useRef();
+
+  const editToDo = () => {
+    dispatch(
+      reduxEditToDo({
```

Comment:
instead of passing the whole new todo object of the editTodo reducer, just pass the id and text to the reducer and inside that reducer just update text of that reducer.

Reply:
newToDo: {
          id: reduxToDoList[index].id,
          text: editField.current.value,
          checked: false,
        },
        
        as currently you are always setting the check=false on edit even if the todo is checked.